packages/video_player/video_player_android/lib/src/android_video_player.dart

```diff
@@ -280,6 +237,130 @@ PlatformVideoViewType _platformVideoViewTypeFromVideoViewType(
   };
 }
 
+/// An instance of a video player, corresponding to a single player ID in
+/// [AndroidVideoPlayer].
+class _PlayerInstance {
+  /// Creates a new instance of [_PlayerInstance] corresponding to the given
+  /// API instance.
+  _PlayerInstance(
+    this._api,
+    this.viewState, {
+    required String eventChannelName,
+  }) {
+    _eventChannel = EventChannel(eventChannelName);
+    _eventSubscription = _eventChannel.receiveBroadcastStream().listen(
+      _onStreamEvent,
+      onError: (Object e) {
+        _eventStreamController.addError(e);
+      },
+    );
+  }
+
+  final VideoPlayerInstanceApi _api;
+  late final EventChannel _eventChannel;
+  final StreamController<VideoEvent> _eventStreamController =
+      StreamController<VideoEvent>.broadcast();
+  late final StreamSubscription<dynamic> _eventSubscription;
+  int _lastBufferPosition = -1;
+
+  final _VideoPlayerViewState viewState;
+
+  Future<void> setLooping(bool looping) {
+    return _api.setLooping(looping);
+  }
+
+  Future<void> play() {
+    return _api.play();
+  }
+
+  Future<void> pause() {
+    return _api.pause();
+  }
+
+  Future<void> setVolume(double volume) {
+    return _api.setVolume(volume);
+  }
+
+  Future<void> setPlaybackSpeed(double speed) {
+    return _api.setPlaybackSpeed(speed);
+  }
+
+  Future<void> seekTo(Duration position) {
+    return _api.seekTo(position.inMilliseconds);
+  }
+
+  Future<Duration> getPosition() async {
+    final PlaybackState state = await _api.getPlaybackState();
+    // TODO(stuartmorgan): Move this logic. This is a workaround for the fact
+    //  that ExoPlayer doesn't have any way to observe buffer position
+    //  changes, so polling is required. To minimize platform channel overhead,
+    //  that's combined with getting the position, but this relies on the fact
+    //  that the app-facing package polls getPosition frequently, which makes
+    //  this fragile (for instance, as of writing, this won't be called while
+    //  the video is paused). It should instead be called on its own timer,
+    //  independent of higher-level package logic.
```

Comment:
I've expanded on the explanation relative to the version removed from VideoPlayer.java, but it's still fundamentally the same hack as the Java version was for now (to minimize behavioral changes in this PR).